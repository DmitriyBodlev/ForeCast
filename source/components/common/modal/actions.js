import { createAction } from 'redux-act';

export const openModal = createAction('openModal');
export const closeModal = createAction('closeModal');
