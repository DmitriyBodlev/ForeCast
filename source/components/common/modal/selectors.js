/**
 * The modal state selectors
 */

import { createSelector } from 'reselect';

const selectModalStore = (state: Object) => state.modal;

const makeSelectModal = () => createSelector(
  selectModalStore,
  (modal: Object) => modal,
);

export {
  selectModalStore,
  makeSelectModal,
};
