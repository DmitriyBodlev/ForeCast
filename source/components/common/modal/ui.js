import styled, { css, keyframes } from 'react-emotion';

export const animation = (props: Object) => {
  const from = props.params.position.split('-')[0];

  const moves = {
    top: keyframes`
      0% { top: -${props.params.height}px }
      100% { top: 0 }
    `,
    bottom: keyframes`
      0% { bottom: -${props.params.height}px }
      100% { bottom: 0 }
    `,
  };

  return moves[from];
};

export const ModalWrapper = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  background: rgba(53, 53, 53, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const Title = styled.span`
`;

export const ModalContent = styled.div`
  padding: 15px;
  position: fixed;
  border-radius: 2px;
  width: ${({params}: Object) => params.width}px;
  height: ${({params}: Object) => params.height}px;
  background-color: grey;
  box-shadow: 0 0 8px 1px black;
  animation: ${(props: Object) => animation(props)} 0.2s linear;
`;

export const Header = styled.div`
  font-weight: bold;
  color: black;
  font-size: 17px;
  align-items: center;
  display: inline-flex;
  width: 100%;
  justify-content: space-between;
`;

export const Footer = styled.div`
  position: absolute;
  bottom: 0px;
  width: 100%;
  border-top: 1px solid;
  height: 55px;
  padding: 10px 15px;
`;

const topLeft = css`
  top: 0;
  left: 0;
`;
const topCenter = css`
  top: 0;
  right: 50%;
  transform: translateX(50%);
`;
const topRight = css`
  top: 0;
  right: 0;
`;

const bottomLeft = css`
  bottom: 0;
  left: 0;
`;

const bottomCenter = css`
  bottom: 0;
  right: 50%;
  transform: translateX(50%);
`;

const bottomRight = css`
  bottom: 0;
  right: 0;
`;

const classStyles = {
  'top-left': topLeft,
  'top-center': topCenter,
  'top-right': topRight,
  'bottom-left': bottomLeft,
  'bottom-center': bottomCenter,
  'bottom-right': bottomRight,
};

export default classStyles;
