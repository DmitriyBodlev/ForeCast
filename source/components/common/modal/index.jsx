import React from 'react';
import { connect } from 'react-redux';
import { closeModal } from './actions';
import styles, { ModalWrapper, Title, ModalContent, Header, Footer } from './ui';


export const HeaderComponent = (props: Object) => (
  <Header>
    <Title>{props.title}</Title>
    <a onClick={props.closeAction} >
      a,sfsafa
    </a>
  </Header>
);

export const FooterComponent = (props: Object) => (
  <Footer>
    {
      props.buttons.map((button: Object, i: number) =>
        <button
          key={i}
          className={button.type}
          onClick={button.action}
        >
          {button.name}
        </button>,
      )
    }
    <button onClick={props.closeAction}>Cansel</button>
  </Footer>
);


export const ModalComponent = (props: Object) => {
  const { options, component } = props.params.modal;
  const closingModal = () => {
    options.closeAction();
    props.closeModal();
  };

  return (
    <ModalWrapper>
      <ModalContent
        className={styles[options.position]}
        params={options}
      >
        <HeaderComponent
          title={options.title}
          closeAction={closingModal} />
        { component }
        {options.controlButtons.length
          && <FooterComponent buttons={options.controlButtons} closeAction={closingModal} />
        }
      </ModalContent>
    </ModalWrapper>
  );
};

export default connect(null, {
  closeModal,
})(ModalComponent);
