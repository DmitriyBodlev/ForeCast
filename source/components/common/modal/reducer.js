import { createReducer } from 'redux-act';
import { $all, $set } from 'plow-js';
import * as Actions from './actions.js';

export const initialState = {
  modal: {},
  isOpened: false,
};

export const openModal = (state: Object, payload: Object) => (
  $all(
    $set('isOpened', true),
    $set('modal', payload),
    state,
  )
);

export const closeModal = (state: Object) => ({
  ...state,
  modal: {},
  isOpened: false,
});

// Can't make closeModal function with plow-js because of
// Converting circular structure to JSON error
// it happened because of initialState.modal.component._owner

export default createReducer({
  [Actions.openModal]: openModal,
  [Actions.closeModal]: closeModal,
}, initialState);
