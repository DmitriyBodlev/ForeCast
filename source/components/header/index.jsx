import React from 'react';
import { connect } from 'react-redux';
import { sendLogOutRequest } from '../../features/auth/actions';
import { changeLngRequest } from '../../features/locales/actions';
import { renderLogoLink } from '../../utils/global-helpers';
import style, { HeaderWrapper, Logo } from './ui';
import { getFirstDataRequest } from '../../features/channels/actions';

export const HeaderComponent = (props: Object) => {
  const { themeName } = props;

  return (
    <HeaderWrapper onClick={props.getFirstDataRequest}>
      <div className={style.brand}>
        <span className={style.logo}>
          <span className={style.helper} />
          <Logo
            src={renderLogoLink(themeName)}
            alt='logo' />
        </span>
        <span className={style.text} />
      </div>
    </HeaderWrapper>
  );
};


export default connect(null, {
  sendLogOutRequest,
  changeLngRequest,
  getFirstDataRequest,
})(HeaderComponent);
