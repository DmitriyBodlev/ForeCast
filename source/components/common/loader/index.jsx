import React from 'react';
import { LoaderWrapper, LoaderContent} from './ui';

export const LoaderComponent = () => {
  return (
    <LoaderWrapper>
      <LoaderContent />
    </LoaderWrapper>
  );
};

export default LoaderComponent;
