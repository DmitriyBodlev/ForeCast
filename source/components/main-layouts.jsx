import React from 'react';
import { connect } from 'react-redux';
import {
  compose,
  pure,
  branch,
  renderComponent,
} from 'recompose';
import { createStructuredSelector } from 'reselect';
import authenticationHOC from '../HOCs/authentication-hoc';
import { makeSelectInitialDataLoading } from '../global-selectors';
import { makeSelectModal } from './common/modal/selectors';
import { makeSelectLoader } from './common/loader/selectors';
import { makeSelectLocale } from '../features/locales/selectors';
import { LayoutsWrapper } from '../ui';
import NavComponent from '../features/menus/navigation';
import SidebarComponent from '../features/menus/sidebar';
import Header from './header';
import Channels from '../features/channels';
import CommonModal from './common/modal';
import LoaderComponent from './common/loader';

const enhance = compose(
  authenticationHOC,
  branch(
    (props: Object) => props.initialDataLoading,
    renderComponent(LoaderComponent),
  pure,
));

export const MainLayouts = (props: Object) => (
  <LayoutsWrapper>
    <Header locale={props.locale} />
    <SidebarComponent />
    <Channels />
    <NavComponent />
    { props.modal.isOpened && <CommonModal params={props.modal} /> }
    { props.loader.isOpened && <LoaderComponent /> }
  </LayoutsWrapper>
);

const mapStateToProps = (state: Object) => (createStructuredSelector({
  initialDataLoading: makeSelectInitialDataLoading(state),
  locale: makeSelectLocale(state),
  modal: makeSelectModal(state),
  loader: makeSelectLoader(state),
}));

export default connect(mapStateToProps)(enhance(MainLayouts));
