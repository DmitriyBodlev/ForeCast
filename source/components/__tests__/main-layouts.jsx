// Standalone runtime for Regenerator-compiled generator and async functions
import 'regenerator-runtime/runtime';

import React from 'react';
import { shallow } from 'enzyme';
import { MainLayouts } from '../main-layouts';
import { HeaderComponent } from '../header';
import { ModalComponent } from '../common/modal';

describe('source/main-layouts.jsx', () => {
  it('should render component with header and modal', () => {
    const wrapper = shallow(<MainLayouts modal={{ isOpened: true }} />);

    const headerComponent = wrapper.find(HeaderComponent);
    const modalComponent = wrapper.find(ModalComponent);

    expect(wrapper).toMatchSnapshot();
    expect(headerComponent).toBeTruthy();
    expect(modalComponent).toBeTruthy();
  });
});
