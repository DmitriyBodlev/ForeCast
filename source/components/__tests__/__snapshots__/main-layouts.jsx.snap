// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source/main-layouts.jsx should render component with header and modal 1`] = `
<styled(Component)>
  <Connect(HeaderComponent) />
  <Connect(withHandlers(Connect(branch(pure(SidebarComponent))))) />
  <Connect(withHandlers(Connect(branch(pure(NavComponent))))) />
  <Connect(ModalComponent)
    params={
      Object {
        "isOpened": true,
      }
    }
  />
</styled(Component)>
`;
